dll/win32/kernel32/client/proc.c

```diff
@@ -451,6 +466,66 @@ WINAPI
 BaseProcessStartup(
     _In_ PPROCESS_START_ROUTINE lpStartAddress)
 {
+    // for winsock inheritance
+    WSAPROTOCOL_INFOW ProtocolInfo;
+    WSADATA wsaData;
+    SOCKET sockDuplicated = INVALID_SOCKET;
+    PBYTE WinsockData = NULL;
+    ULONG sockindex = 0;
+    int sockloop;
+    int nStatus;
+    HMODULE winsock_module = NULL;
+    PPEB Peb = NtCurrentPeb();
+    typedef int (WINAPI *wsstcall)(DWORD, LPWSADATA);
+    typedef int (WINAPI *wsskcall)(int, int, int, LPWSAPROTOCOL_INFOW, GROUP, DWORD);
+    wsstcall DynWSAStartup;
+    wsskcall DynWSASocket;
+
+    RtlAcquirePebLock();
+    WinsockData = Peb->SparePtr2; // retreive PEB stored winsock inheritance data
+    sockindex = Peb->SpareUlong;
+    RtlReleasePebLock();
+
+    if (sockindex > 0) {
+        DPRINT("BaseProcessStartup sockindex retreived:%lx\n", sockindex);
+        DPRINT("BaseProcessStartup WinsockData pointer retreived:%p\n", WinsockData);
+	    winsock_module = LoadLibrary("WS2_32.DLL"); 
+  	    if (winsock_module)
+  	    {
+    		DynWSAStartup = (wsstcall)GetProcAddress(winsock_module, "WSAStartup");
+    		DynWSASocket = (wsskcall)GetProcAddress(winsock_module, "WSASocketW");
+	    } else {
+ 		    DPRINT1("BaseProcessStartup: winsock module faled to load it into child process.\n");
+	    }
+        if (winsock_module) 
+        {
+	        if ( DynWSASocket && DynWSAStartup ) 
+            {
+    	        if ((nStatus = DynWSAStartup(0x202,&wsaData)) == 0)
+    	        {
+		            for (sockloop = 0; sockloop < sockindex; sockloop = sockloop + 1) 
+                    {
+	            	    memcpy(&ProtocolInfo, &WinsockData[sockloop * ( sizeof(ProtocolInfo) )] , sizeof(ProtocolInfo));
```

Comment:
```suggestion
	            	    memcpy(&ProtocolInfo, &WinsockData[sockloop * sizeof(ProtocolInfo)], sizeof(ProtocolInfo));
```
Looks to my eyes as if we could avoid one level of braces here.